src/main/java/gift/main/controller/ProductController.java

```diff
@@ -0,0 +1,74 @@
+package gift.main.controller;
+
+import gift.main.dto.ListProductResponse;
+import gift.main.dto.ProductRequest;
+import gift.main.dto.Response;
+import gift.main.dto.SingleProductResponse;
+import gift.main.entity.Product;
+import gift.main.handler.MapToProductTransformer;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+@RestController
+public class ProductController {
+    private final Map<Long, Product> producstRepository = new HashMap<>();
+    private final AtomicLong idGenerator = new AtomicLong();
+
+    @GetMapping("/products")
+    public Response provideProductAll() {
+        if (producstRepository.isEmpty()) {
+            return new Response(200, "현재 모든 물건이 비어있습니다.");
+        }
+
+        List<Product> productList = new ArrayList<>();
+        for (Map.Entry<Long, Product> productEntry : producstRepository.entrySet()) {
+            productList.add(productEntry.getValue());
+        }
+
+        return new ListProductResponse(200, "ok", productList);
+    }
+
+    @GetMapping("/product")

```

Comment:
RESTful 하게 많은 리소스들 중에서 해당 id 값을 찾는 다는 의미로 아래 PATH 로 변경해보는 것은 어떨까요?!
```suggestion
@GetMapping("/products/{id}")
```

fyi; [RESTful API 관련 문서](https://meetup.nhncloud.com/posts/92)